### PR TITLE
add no-identical-title rule (fixes #7)

### LINKED ISSUE #7: `no-identical-title` rule
Disallow tests with identical titles as it makes it hard to differentiate.

Might be better off in AVA itself: https://github.com/sindresorhus/ava/issues/517

### PR DESCRIPTION
**Pull request by jfmengels**
Adds no-identical-title rule (fixes #7)

Let me know if I need to change anything ;)

### CODE REVIEW

File: docs/rules/no-identical-rule.md [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+});
+```
+
+
+## Pass
+
+```js
+const test = require('ava');
+
+test((t) => {
```
Comment (sindresorhus) [inline docs/rules/no-identical-rule.md]:
```js
test(t => {
```

**Comment by wyze** [comment]
The documentation filename is `no-identical-rule.md` and should be `no-identical-title.md`. This would match the name of the rule and would have the correct link in the readme.

**Comment by jfmengels** [comment]
Good catch @wyze!
Updated it to use `createAvaRule`.
I'm putting this one on hold until #13 gets merged, as I now use some util method introduced there. (Or rather, this PR includes the commit from #13, and should get swallowed once #13 is merged)

**Comment by jfmengels** [comment]
Updated!
It's not using the utils method anymore introduced in #13 (since it got removeed from that PR just now anyway), so both can be merged in whatever order.

**Comment by sindresorhus** [comment]
Could you fix the merge conflict?

**Comment by jfmengels** [comment]
Done ;)

**Comment by jfmengels** [comment]
\o/

**Comment by sindresorhus** [comment]
celebration

**Pull request merged**